pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -105,22 +106,49 @@ func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) (model
 		Hosts: []string{"*"},
 	})
 
-	virtualService := &networking.VirtualService{
-		Hosts:    []string{"*"},
-		Gateways: []string{model.IstioIngressGatewayName},
+	gatewayConfig := model.Config{
+		ConfigMeta: model.ConfigMeta{
+			Type:      model.Gateway.Type,
+			Group:     model.Gateway.Group,
+			Version:   model.Gateway.Version,
+			Name:      ingress.Name + "-" + model.IstioIngressGatewayName,
+			Namespace: model.IstioIngressNamespace,
+			Domain:    domainSuffix,
+		},
+		Spec: gateway,
 	}
 
-	var httpRoutes []*networking.HTTPRoute
+	return gatewayConfig
+}
+
+func ConvertIngressVirtualService(ingress v1beta1.Ingress, domainSuffix string, ingressByHost map[string]*model.Config) {
+	// Ingress allows a single host - if missing '*' is assumed
+	// We need to merge all rules with a particular host across
+	// all ingresses, and return a separate VirtualService for each
+	// host.
+
 	for _, rule := range ingress.Spec.Rules {
 		if rule.HTTP == nil {
 			log.Infof("invalid ingress rule %s:%s for host %q, no paths defined", ingress.Namespace, ingress.Name, rule.Host)
 			continue
 		}
 
+		host := rule.Host
+		namePrefix := strings.Replace(host, ".", "-", -1)
```

Comment:
even though it's trivial, wrapping this in a helper function will document why this replace is needed.

Reply:
And likely make the code harder to read (context switch). I can add a comment if it is not obvious ( dot not valid in names)

Reply:
why would namePrefix := replaceInvalidDotsInName(host) be harder to read? this is self commenting.

Reply:
It is not - it doesn't say how they are replaced, why they are invalid ( they are not invalid in a hostname ), and in the unlikely event it will be reused accidentally can lead to bigger problems. It's a one liner. Maybe 'replaceAllDotsWithMinus' would be self commenting. In go it is considered ok to duplicate small pieces of code.